src/core/logical_layers/renderers/BivariateRenderer/BivariateRenderer.tsx

```diff
@@ -425,6 +425,7 @@ export class BivariateRenderer extends LogicalLayerDefaultRenderer {
     }
 
     this.cleanPopup();
+    this.resetFeatureStates?.();
```

Comment:
![image](https://github.com/user-attachments/assets/98f57739-09ff-47a8-aca3-3374af9d4d65)

according to the fact that at the code we have this method of 'this' added dynamically, I can be wrong, but the function might be undefined (that is why we use '?') 

a new code and also other places where this method is called do not have any handling of potential errors when function is absent 

actually what we are doing now is reseting feature states only if 'resetFeatureStates' is defined

I'd suggest doing smth like 

`this.resetFeatureStates?.() ?? console.error('No resetFeatureStates function is defined');`

or handle this error in other better ways to cover both decisions of if statement added here. If we already have one - great. Just in case we don't handle that

Reply:
@amurKontur from what I see - all the usages of `resetFeatureStates` are covered by conditional chaining (?.). 

Don't think think adding a new error message is really needed here - it's just a cleanup function, and if resetFeatureStates is not defined, there's nothing to clean up.
Plus, `resetFeatureStates()` does not return any value, so the result of condition `this.resetFeatureStates?.()` will always be false.

Reply:
As for `onMapZoomHandlers` - it looks like they're not connected to anything and do literally nothing. @Natallia-Harshunova could you add a TODO comment about those?